### ISSUE TITLE
PostgreSQL: `CREATE INDEX` with cast to schema-qualified type with string argument fails to parse

### ISSUE THREAD

**Issue by abdelfattahradwan** [sqlfluff/sqlfluff#7820]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A `CREATE INDEX` statement containing a cast to a schema-qualified type with a string argument (e.g. `col::schema.type('arg=value')`) is reported as an unparsable section. The syntax is accepted by PostgreSQL when the type is registered by an extension, so this looks like a gap in the postgres dialect grammar.

### Expected Behaviour

The statement parses with no `PRS` error.

### Observed Behaviour

The entire CREATE INDEX statement is flagged as unparsable:

```plaintext
L: 1 | P: 1 | PRS | Line 1, Position 1: Found unparsable section: 'create index ...'
```

### How to reproduce

Lint the following with `sqlfluff lint --dialect postgres`:

```sql
create index table_a_idx_a on schema_a.table_a
    using bm25 (col_a, (col_a::pdb.unicode_words('alias=col_a_alias')))
    with (key_field = 'col_a');
```

### Dialect

postgres

### Version

sqlfluff, version 4.1.0

Python 3.14.3

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @abdelfattahradwan  thnaks for opening this issue.

Question: is this  a standard postgres syntax? Can you include here the link to the postgres syntax reference pls?

**Comment by abdelfattahradwan** [comment]
> Hi [@abdelfattahradwan](https://github.com/abdelfattahradwan) thnaks for opening this issue.
> 
> Question: is this a standard postgres syntax? Can you include here the link to the postgres syntax reference pls?

Yep, it is standard PostgreSQL syntax. It is simply a schema-qualified typecast with a type modifier.

`(col_a::pdb.unicode_words('alias=col_a_alias'))` breaks down as:

- `col_a`
- - source expression/column
- `::`
- - PostgreSQL explicit cast operator
- `pdb.unicode_words`
- - schema-qualified destination type
- `('alias=col_a_alias')`
- - type modifier (typmod)

This is the same syntax category as built-in PostgreSQL examples such as: `col_a::varchar(32)` or `price::numeric(10,2)` where `(32)` and `(10,2)` are also type modifiers. But in this case, ParadeDB's `pdb.unicode_words` accepts a typmod string instead of numeric parameters.

https://www.postgresql.org/docs/current/sql-expressions.html#SQL-SYNTAX-TYPE-CASTS

https://www.postgresql.org/docs/current/sql-createtype.html

https://www.postgresql.org/docs/current/sql-createcast.html
